web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -0,0 +1,287 @@
+import { useWorkspaceContext } from 'teleterm/ui/Documents';
+import { useWorkspaceLoggedInUser } from 'teleterm/ui/hooks/useLoggedInUser';
+import { useConnectMyComputerContext } from 'teleterm/ui/ConnectMyComputer';
+
+import { retryWithRelogin } from 'teleterm/ui/utils';
+import { routing } from 'teleterm/ui/uri';
+
+import {
+  ConnectServerActionButton,
+  ConnectKubeActionButton,
+  ConnectDatabaseActionButton,
+} from './actionButtons';
+import { useResourcesContext } from './resourcesContext';
+
+interface UnifiedResourcesProps {
+  clusterUri: uri.ClusterUri;
+}
+
+export function UnifiedResources(props: UnifiedResourcesProps) {
+  const appContext = useAppContext();
+  const { onResourcesRefreshRequest } = useResourcesContext();
+
+  const [params, setParams] = useState<UnifiedResourcesQueryParams>({
+    sort: { fieldName: 'name', dir: 'ASC' },
+  });
+
+  const { documentsService, rootClusterUri } = useWorkspaceContext();
+  const loggedInUser = useWorkspaceLoggedInUser();
+  const { canUse: hasPermissionsForConnectMyComputer, agentCompatibility } =
+    useConnectMyComputerContext();
+
+  const isRootCluster = props.clusterUri === rootClusterUri;
+  const canAddResources = isRootCluster && loggedInUser?.acl?.tokens.create;
+
+  const canUseConnectMyComputer =
+    isRootCluster &&
+    hasPermissionsForConnectMyComputer &&
+    agentCompatibility === 'compatible';
+
+  const { fetch, resources, attempt } = useUnifiedResourcesFetch({
+    fetchFunc: useCallback(
+      async (paginationParams, signal) => {
+        const response = await retryWithRelogin(
+          appContext,
+          props.clusterUri,
+          () =>
+            appContext.resourcesService.listUnifiedResources(
+              {
+                clusterUri: props.clusterUri,
+                searchAsRoles: false,
+                sortBy: {
+                  isDesc: params.sort.dir === 'DESC',
+                  field: params.sort.fieldName,
+                },
+                search: params.search,
+                kindsList: params.kinds,
+                query: params.query,
+                pinnedOnly: params.pinnedOnly,
+                startKey: paginationParams.startKey,
+                limit: paginationParams.limit,
+              },
+              signal
+            )
+        );
+
+        return {
+          startKey: response.nextKey,
+          agents: response.resources,
+          totalCount: response.resources.length,
+        };
+      },
+      [
+        appContext,
+        params.kinds,
+        params.pinnedOnly,
+        params.query,
+        params.search,
+        params.sort.dir,
+        params.sort.fieldName,
+        props.clusterUri,
+      ]
+    ),
+  });
+
+  useEffect(() => {
+    const { cleanup } = onResourcesRefreshRequest(() =>
+      fetch({
+        force: true,
+        fromStart: true,
+      })
+    );
+    return cleanup;
+  }, [onResourcesRefreshRequest, fetch]);
+
+  return (
+    <SharedUnifiedResources
```

Comment:
`SharedUnifiedResources` uses `web/packages/shared/components/UnifiedResources/unifiedStyles.css` underneath. That `min-width: 800px` makes this look bad in Connect which can be much more narrower than that. If this is something that the Web UI needs, I think it should be set through another stylesheet that's imported only in the Web UI.

Reply:
I will check if we even need that `min-width` for Web UI. Will address that soon in a PR that will be backported.